frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -4,15 +4,14 @@ import axios from "axios";
 import {
     lookupTableApi,
     LookupTable,
-    ValueSet,
     LookupTables,
 } from "../network/api/LookupTableApi";
 import { showError } from "../components/AlertNotifications";
 
 export const generateUseLookupTable =
-    <T>(tableName: LookupTables) =>
+    <T>(tableName: LookupTables, dataSetName: string | null = null) =>
     () => {
-        return useLookupTable<T>(tableName);
+        return useLookupTable<T>(tableName, dataSetName);
```

Comment:
What's the benefit of this proxy function? I was looking at `useLookupTable` and I think this could be useful as the null checker and error thrower so you don't have to muddy up the effect hook with a conditional. Just an observation, still curious about this function's intent.

Reply:
@kevinhaube while `useLookupTable` is our primary hook for (mostly) generalized lookup table access, the main idea behind `generateUseLookupTable` is that it sets up the `useLookupTable` once upon page load, so that we can (hopefully) get some performance gains by reducing updates to the reactive model. Tagging @doug-s-nava and @ahay-agile6 who can (hopefully) explain this architectural pattern a little better, as I am a react newb

Reply:
Interested to hear more about this! I know React docs tout `useMemo` and `useCallback` as ways of helping with performance gains related to reactive models. Wonder if we could brainstorm a clearer approach. Short of that, some documentation about this would help 👍🏼

Reply:
In fact, just had a thought that might work... 
```typescript
/* Run once and memoize values */
export const useMemoizedLookupTable = useMemo<T>(() => 
    return useLookupTable<T>(tableName, dataSetName), 
    []
)
```
This will memoize the return value of the hook, and if the empty dependency array gives you trouble about `tableName` or `dataSetName`, just `eslint-ignore` it, it's the linter having an opinion against the empty deps feature in React, not a legit typescript issue. 

This also maintains its integrity. Using `use` to begin a function name will tell React to check that we're only using it inside components and other hooks. 👍🏼